trade-finance-manager-api/src/v1/middleware/validate-put-deal-cancellation-payload.test.ts

```diff
@@ -0,0 +1,80 @@
+import { MAX_CHARACTER_COUNT, TfmDealCancellation } from '@ukef/dtfs2-common';
+import httpMocks from 'node-mocks-http';
+import { HttpStatusCode } from 'axios';
+import { validatePutDealCancellationPayload } from './validate-put-deal-cancellation-payload';
+
+describe('validatePostPaymentPayload', () => {
+  const getHttpMocks = () => httpMocks.createMocks();
+
+  const aValidPayload = (): Partial<TfmDealCancellation> => ({
+    reason: 'x'.repeat(MAX_CHARACTER_COUNT),
+    bankRequestDate: new Date().valueOf(),
+    effectiveFrom: new Date().valueOf(),
+  });
+
+  const invalidPayloads = [
+    {
+      description: 'the payload is undefined',
+      payload: undefined,
+    },
+    {
+      description: "the 'reason' is not a string",
+      payload: {
+        ...aValidPayload(),
+        reason: 1234,
+      },
+    },
+    {
+      description: `the 'reason' is over ${MAX_CHARACTER_COUNT} characters`,
+      payload: {
+        ...aValidPayload(),
+        reason: 'x'.repeat(MAX_CHARACTER_COUNT + 1),
+      },
+    },
+    {
+      description: "the 'effectiveFrom' is a string",
+      payload: {
+        ...aValidPayload(),
+        effectiveFrom: new Date().toString(),
+      },
+    },
+    {
+      description: "the 'bankRequestDate' is a string",
+      payload: {
+        ...aValidPayload(),
+        bankRequestDate: new Date().toString(),
+      },
+    },
+  ];
```

Comment:
Seems we have both central API and TFM API testing the same validation/response?

Reply:
Yep, this is intentional. My understanding is each microservice should validate anything it recieves & and not rely on this being done elsewhere